server/src/test/java/org/elasticsearch/gateway/GatewayMetaStateTests.java

```diff
@@ -98,11 +100,37 @@ public void testCustomMetadataNoChange() {
         }
     }
 
+    public void testCustomMetadata_appliesUpgraders() {
+        CustomMetadata1 originalCustom1 = new CustomMetadata1("old data");
+        CustomMetadata1 upgradedCustom1 = new CustomMetadata1("new data");
+        CustomMetadata2 custom2 = new CustomMetadata2("some data");
+        Metadata originalMetadata = randomMetadata(originalCustom1, custom2);
```

Comment:
I have troubles understading this test: it seems we are testing randomly either one custom metadata type or the other?

Reply:
Yeah, I can see how this is confusing, sorry! In fact, `randomMetadata` isn't introducing any randomness to do with the customs, only in other irrelevant stuff. I'll replace it with something more explicitly showing what needs to be created for the test, and add a comment or two.